File: ios/Classes/FlutterAppsOnAirPlugin.swift
import Flutter
import UIKit
import AppsOnAir

public class FlutterAppsOnAirPlugin: NSObject, FlutterPlugin {
    
    private var appId: String = ""
    let appsOnAirService = AppsOnAirServices()
    
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "flutter_apps_on_air", binaryMessenger: registrar.messenger())
    let instance = FlutterAppsOnAirPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
      if ("setAppID" == call.method) {
            if let args = call.arguments as? Dictionary<String, Any>,
                 let appId = args["AppId"] as? String, let showNativeUI  = args["showNativeUI"] as? Bool{
                      do{
                          appsOnAirService.setAppId(appId, showNativeUI)
                          print("iOS application ID \(appId) and show native UI \(showNativeUI)")
                      }catch let error {
                          print("Failed to load: \(error.localizedDescription)")
                      }
                      result(true)
                  } else {
                      result(false)
            }
    }
  }
}

File: android/src/main/kotlin/com/shivam/flutter_apps_on_air/FlutterAppsOnAirPlugin.kt
package com.shivam.flutter_apps_on_air

import androidx.annotation.NonNull
import com.appsonair.AppsOnAirServices

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** FlutterAppsOnAirPlugin */
class FlutterAppsOnAirPlugin: FlutterPlugin, MethodCallHandler {
  /// The MethodChannel that will the communication between Flutter and native Android
  ///
  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
  /// when the Flutter Engine is detached from the Activity
  private lateinit var channel : MethodChannel

  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "flutter_apps_on_air")
    channel.setMethodCallHandler(this)
  }

  override fun onMethodCall(call: MethodCall, result: Result) {
    if (call.method == "setAppID") {
      // Retrieve the argument as a nullable Boolean
      val showNativeUI: Boolean = call.argument<Boolean>("showNativeUI") ?: false

      AppsOnAirServices.setAppId(call.argument("AppId"), showNativeUI)
      result.success(true)
    }
      else {
        result.success(false)
      }

  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }
}
